core/src/lightclient/mod.rs

```diff
@@ -340,6 +349,32 @@ impl StarknetClient for StarknetClientImpl {
         }
     }
 
+    /// Get the number of transactions in a block given a block hash.
+    /// The number of transactions in a block.
+    /// # Arguments
+    /// * `hash(H256)` - The block hash.
+    /// # Returns
+    ///
+    ///  * `transaction_count(U256)` - The number of transactions.
+    ///
+    /// `Ok(Option<U256>)` if the operation was successful.
+    /// `Err(LightClientError)` if the operation failed.
+    async fn block_transaction_count_by_hash(
+        &self,
+        hash: H256,
+    ) -> Result<Option<U256>, LightClientError> {
+        let starknet_block_id = ethers_block_id_to_starknet_block_id(BlockId::Hash(hash))?;
```

Comment:
nice addition, could you adapt this change to block_transaction_count_by_number?